testsuite/testcases/src/compatibility_test.rs

```diff
@@ -33,85 +25,17 @@ impl Test for SimpleValidatorUpgrade {
-) -> Result<Option<TxnStats>> {
-    let mut upgrade_stats = vec![];
-    while !done.load(Ordering::Relaxed) {
-        info!("stat_gather_task some traffic...");
-        let upgrading_stats = emitter
-            .clone()
-            .emit_txn_for(
-                source_account.clone(),
-                emit_job_request.clone(),
-                upgrade_traffic_chunk_duration,
-            )
-            .await?;
-        info!("stat_gather_task some stats: {}", &upgrading_stats);
-        upgrade_stats.push(upgrading_stats);
-    }
-    let statsum = upgrade_stats.into_iter().reduce(|a, b| &a + &b);
-    Ok(statsum)
-}
-
-fn upgrade_and_gather_stats(
+fn upgrade(
     ctxa: NetworkContextSynchronizer,
     // upgrade args
     validators_to_update: &[PeerId],
     version: &Version,
     wait_until_healthy: bool,
     delay: Duration,
     max_wait: Duration,
-    // traffic args
-    nodes: &[PeerId],
-) -> Result<Option<TxnStats>> {
-    let upgrade_done = Arc::new(AtomicBool::new(false));
-    let emitter_ctx = ctxa.clone();
-    let mut stats_result: Result<Option<TxnStats>> = Ok(None);
+) -> Result<()> {
     let mut upgrade_result: Result<()> = Ok(());
     tokio_scoped::scope(|scopev| {
-        // emit trafic and gather stats
-        scopev.spawn(async {
-            info!("upgrade_and_gather_stats traffic thread start");
-            let (emitter, emit_job_request, source_account) = {
-                let mut ctx_locker = emitter_ctx.ctx.lock().await;
-                let ctx = ctx_locker.deref_mut();
-                let emit_job_request = ctx.emit_job.clone();
-                let rng = SeedableRng::from_rng(ctx.core().rng()).unwrap();
-                let (emitter, emit_job_request) = match create_emitter_and_request(
-                    ctx.swarm.clone(),
-                    emit_job_request,
-                    nodes,
-                    rng,
-                )
-                .await
-                {
-                    Ok(parts) => parts,
-                    Err(err) => {
-                        stats_result = Err(err);
-                        return;
-                    },
-                };
-                let source_account = ctx.swarm.read().await.chain_info().root_account;
-                (emitter, emit_job_request, source_account)
-                // release lock on network context
-            };
-            let upgrade_traffic_chunk_duration = Duration::from_secs(15);
-            info!("upgrade_and_gather_stats traffic thread 1");
-            stats_result = stat_gather_task(
-                emitter,
-                emit_job_request,
-                source_account,
-                upgrade_traffic_chunk_duration,
-                upgrade_done.clone(),
-            )
-            .await;
-            info!("upgrade_and_gather_stats traffic thread done");
-        });
         // do upgrade
         scopev.spawn(async {
             info!("upgrade_and_gather_stats upgrade thread start");
```

Comment:
Nit remove now that func name changed and don't need to spawn thread anymore

Reply:
Let's leave it as we still need the spawn.. updated the message accordingly.